tests/test_sft_trainer.py

```diff
@@ -634,3 +633,121 @@ def test_run_with_additional_callbacks():
+
+def test_run_with_bad_experimental_metadata():
+    """Ensure that train() throws error with bad experimental metadata"""
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        train_args = copy.deepcopy(TRAIN_ARGS)
+        train_args.output_dir = tempdir
+        model_args = copy.deepcopy(MODEL_ARGS)
+
+        metadata = "deadbeef"
+
+        with pytest.raises(ValueError):
+            sft_trainer.train(
+                model_args,
+                DATA_ARGS,
+                train_args,
+                PEFT_PT_ARGS,
+                additional_callbacks=[TrainerCallback()],
+                exp_metadata=metadata
+            )
+
+def test_run_with_good_experimental_metadata():
+    """Ensure that train() can work with good experimental metadata"""
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        train_args = copy.deepcopy(TRAIN_ARGS)
+        train_args.output_dir = tempdir
+        model_args = copy.deepcopy(MODEL_ARGS)
+
+        metadata = {"dead": "beef"}
+
+        sft_trainer.train(
+            model_args,
+            DATA_ARGS,
+            train_args,
+            PEFT_PT_ARGS,
+            additional_callbacks=[TrainerCallback()],
+            exp_metadata=metadata
+        )
+
+#### Tracker subsystem checks
+
+def test_run_with_bad_tracker_config():
+    """Ensure that train() raises error with bad tracker configs"""
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        train_args = copy.deepcopy(TRAIN_ARGS)
+        train_args.output_dir = tempdir
+        model_args = copy.deepcopy(MODEL_ARGS)
+
+        with pytest.raises(ValueError):
+            sft_trainer.train(
+                model_args,
+                DATA_ARGS,
+                train_args,
+                PEFT_PT_ARGS,
+                tracker_configs="NotSupposedToBeHere",
+            )
+
+def test_run_with_bad_tracker_name():
+    """Ensure that train() raises error with bad tracker name"""
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        train_args = copy.deepcopy(TRAIN_ARGS)
+        train_args.output_dir = tempdir
+        model_args = copy.deepcopy(MODEL_ARGS)
+
+        train_args.trackers = ["NotAValidTracker"]
+
+        with pytest.raises(ValueError):
+            sft_trainer.train(
+                model_args,
+                DATA_ARGS,
+                train_args,
+                PEFT_PT_ARGS,
+            )
+
+def test_run_with_good_tracker_name_but_no_args():
+    """Ensure that train() raises error with bad tracker name"""
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        train_args = copy.deepcopy(TRAIN_ARGS)
+        train_args.output_dir = tempdir
+        model_args = copy.deepcopy(MODEL_ARGS)
+
+        train_args.trackers = ["aim"]
+
+        with pytest.raises(ValueError):
+            sft_trainer.train(
+                model_args,
+                DATA_ARGS,
+                train_args,
+                PEFT_PT_ARGS,
+            )
+
+def test_run_with_good_tracker_name():
```

Comment:
this is an end to end test which requires `aim` to be installed which is an optional dependency if we go about testing this it would need us to mock the whole subsystem of AimCallback and the AimStackTracker which I feel is too much delta at this point. WDYT 

cc @ashokponkumar

Reply:
Also another test case when passing both the trackers aim and file with proper test inputs asserting a success. :) Thanks.

Reply:
done